public/components/integrations/components/setup_integration.tsx

```diff
@@ -118,10 +119,52 @@ const suggestDataSources = async (type: string): Promise<Array<{ label: string }
   }
 };
 
-const findTemplate = async (integrationTemplateId: string) => {
-  const http = coreRefs.http!;
-  const result = await http.get(`${INTEGRATIONS_BASE}/repository/${integrationTemplateId}`);
-  return result;
+const runQuery = async (
+  query: string,
+  trackProgress: (step: number) => void
+): Promise<Result<object>> => {
+  // Used for polling
+  const sleep = (ms: number) => {
+    return new Promise((resolve) => setTimeout(resolve, ms));
+  };
+
+  try {
+    console.log('- Posting query');
+    const http = coreRefs.http!;
+    const queryId = (
+      await http.post(CONSOLE_PROXY, {
+        body: JSON.stringify({ query, lang: 'sql' }),
+        query: {
+          path: '_plugins/_async_query',
+          method: 'POST',
+        },
+      })
+    ).queryId;
+    while (true) {
+      const poll = await http.post(CONSOLE_PROXY, {
+        body: '{}',
+        query: {
+          path: '_plugins/_async_query/' + queryId,
+          method: 'GET',
+        },
+      });
+      console.log('- Poll:', poll);
+      if (poll.status === 'PENDING') {
+        trackProgress(1);
+      } else if (poll.status === 'RUNNING') {
+        trackProgress(2);
+      } else if (poll.status === 'SUCCESS') {
+        trackProgress(3);
+        return { ok: true, value: poll };
+      } else if (poll.status === 'FAILURE') {
+        return { ok: false, error: new Error('FAILURE status', { cause: poll }) };
+      }
+      await sleep(3000);
+    }
+  } catch (err: any) {
+    console.error(err);
+    return { ok: false, error: err };
```

Comment:
i think this works but usually it's `error, loading, data`. If loading is true then display loading, else if error is not null then display error, otherwise data is useable. for example like https://swr.vercel.app/#overview